src/main/java/seedu/address/logic/commands/ScheduleCommand.java

```diff
@@ -67,23 +81,25 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_INVALID_NAME);
         }
 
-        // Check if the schedule time is valid (on the hour and within weekday working hours)
-        if (!isOnTheHour(schedule.dateTime) || !isWithinWorkingHours(schedule.dateTime)) {
-            throw new CommandException(MESSAGE_INVALID_TIME);
-        }
+        for (Schedule schedule : scheduleSet) {
+            // Check if the schedule time is valid (on the hour and within weekday working hours)
+            if (!isOnTheHour(schedule.getDateTime()) || !isWithinWorkingHours(schedule.getDateTime())) {
```

Comment:
I like that the code clearly separates the validation checks, making it easy to understand what each condition is verifying!